Acceptance tests can leak hbars for failures during startup

### Description

Due to a race condition with verifying a receipt against one node and querying against another node, acceptance tests can sometimes fail on startup inside the bean constructor and fail to run the cleanup method to delete and refund.

### Steps to reproduce

1. Run acceptance tests
2. Hope the race condition occurs

### Additional context

```
[2025-07-01T06:42:28.417Z] 2025-07-01T06:42:28.065Z  INFO ForkJoinPool-1-worker-1 o.h.m.t.e.a.c.SDKClient Created operator account 0.0.9319680 with public key 302a300506032b6570032100cee10aecee86f27792737ddd68cc1b9e78f9195032afd53f81ec76ba065ec49f 
[2025-07-01T06:42:33.794Z] 2025-07-01T06:42:33.431Z  WARN ForkJoinPool-1-worker-1 o.s.w.c.s.GenericWebApplicationContext Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'accountClient' defined in URL [jar:file:/app/test.jar!/org/hiero/mirror/test/e2e/acceptance/client/AccountClient.class]: Failed to instantiate [org.hiero.mirror.test.e2e.acceptance.client.AccountClient]: Constructor threw exception [2025-07-01T06:42:33.799Z] 2025-07-01T06:42:33.483Z ERROR ForkJoinPool-1-worker-1 o.s.b.SpringApplication Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'accountClient' defined in URL [jar:file:/app/test.jar!/org/hiero/mirror/test/e2e/acceptance/client/AccountClient.class]: Failed to instantiate [org.hiero.mirror.test.e2e.acceptance.client.AccountClient]: Constructor threw exception
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1395)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1222)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1188)
[2025-07-01T06:42:33.799Z] 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1123)
[2025-07-01T06:42:33.799Z] 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
[2025-07-01T06:42:33.799Z] 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.test.context.SpringBootContextLoader.lambda$loadContext$3(SpringBootContextLoader.java:144)
[2025-07-01T06:42:33.799Z] 	at org.springframework.util.function.ThrowingSupplier.get(ThrowingSupplier.java:58)
[2025-07-01T06:42:33.799Z] 	at org.springframework.util.function.ThrowingSupplier.get(ThrowingSupplier.java:46)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.SpringApplication.withHook(SpringApplication.java:1461)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.test.context.SpringBootContextLoader$ContextLoaderHook.run(SpringBootContextLoader.java:563)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:144)
[2025-07-01T06:42:33.799Z] 	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:110)
[2025-07-01T06:42:33.799Z] 	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:225)
[2025-07-01T06:42:33.799Z] 	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:152)
[2025-07-01T06:42:33.799Z] 	at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:130)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.spring.TestContextAdaptor.create(TestContextAdaptor.java:42)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.spring.SpringFactory.start(SpringFactory.java:121)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.core.runner.Runner.buildBackendWorlds(Runner.java:135)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.core.runner.Runner.runPickle(Runner.java:71)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.lambda$runTestCase$4(CucumberEngineExecutionContext.java:114)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:136)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:136)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.runTestCase(CucumberEngineExecutionContext.java:111)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:168)
[2025-07-01T06:42:33.799Z] 	at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:90)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:156)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:146)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:144)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:143)
[2025-07-01T06:42:33.799Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.exec(ForkJoinPoolHierarchicalTestExecutorService.java:274)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.execSync(ForkJoinPoolHierarchicalTestExecutorService.java:247)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:159)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:160)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:146)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:144)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:143)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.exec(ForkJoinPoolHierarchicalTestExecutorService.java:274)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.execSync(ForkJoinPoolHierarchicalTestExecutorService.java:247)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:159)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:160)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:146)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:144)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:143)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.exec(ForkJoinPoolHierarchicalTestExecutorService.java:274)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.join(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.joinConcurrentTasksInReverseOrderToEnableWorkStealing(ForkJoinPoolHierarchicalTestExecutorService.java:201)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:167)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:160)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:146)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:144)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:143)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.exec(ForkJoinPoolHierarchicalTestExecutorService.java:274)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.join(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.joinConcurrentTasksInReverseOrderToEnableWorkStealing(ForkJoinPoolHierarchicalTestExecutorService.java:201)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:167)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:160)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:146)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:144)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:143)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
[2025-07-01T06:42:33.800Z] 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.exec(ForkJoinPoolHierarchicalTestExecutorService.java:274)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
[2025-07-01T06:42:33.800Z] Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.hiero.mirror.test.e2e.acceptance.client.AccountClient]: Constructor threw exception
[2025-07-01T06:42:33.800Z] 	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:223)
[2025-07-01T06:42:33.800Z] 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:145)
[2025-07-01T06:42:33.800Z] 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
[2025-07-01T06:42:33.800Z] 	... 104 common frames omitted
[2025-07-01T06:42:33.800Z] Caused by: com.hedera.hashgraph.sdk.PrecheckStatusException: failed pre-check with the status `INVALID_ACCOUNT_ID`
[2025-07-01T06:42:33.800Z] 	at com.hedera.hashgraph.sdk.Executable$GrpcRequest.mapStatusException(Executable.java:968)
[2025-07-01T06:42:33.800Z] 	at com.hedera.hashgraph.sdk.Executable.execute(Executable.java:451)
[2025-07-01T06:42:33.800Z] 	at com.hedera.hashgraph.sdk.Query.execute(Query.java:31)
[2025-07-01T06:42:33.800Z] 	at com.hedera.hashgraph.sdk.Executable.execute(Executable.java:353)
[2025-07-01T06:42:33.800Z] 	at com.hedera.hashgraph.sdk.Query.execute(Query.java:31)
[2025-07-01T06:42:33.800Z] 	at org.hiero.mirror.test.e2e.acceptance.client.AbstractNetworkClient.lambda$executeQuery$2(AbstractNetworkClient.java:85)
[2025-07-01T06:42:33.800Z] 	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:357)
[2025-07-01T06:42:33.800Z] 	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:230)
[2025-07-01T06:42:33.800Z] 	at org.hiero.mirror.test.e2e.acceptance.client.AbstractNetworkClient.executeQuery(AbstractNetworkClient.java:85)
[2025-07-01T06:42:33.800Z] 	at org.hiero.mirror.test.e2e.acceptance.client.AbstractNetworkClient.getBalance(AbstractNetworkClient.java:191)
[2025-07-01T06:42:33.800Z] 	at org.hiero.mirror.test.e2e.acceptance.client.AbstractNetworkClient.getBalance(AbstractNetworkClient.java:186)
[2025-07-01T06:42:33.800Z] 	at org.hiero.mirror.test.e2e.acceptance.client.AccountClient.<init>(AccountClient.java:47)
[2025-07-01T06:42:33.800Z] 	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
[2025-07-01T06:42:33.800Z] 	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:210)
[2025-07-01T06:42:33.800Z] 	... 106 common frames omitted
```

### Hedera network

other

### Version

v0.133.0

### Operating system

None

--- a/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/AccountClient.java
+++ b/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/AccountClient.java
@@ -44,8 +44,16 @@ public class AccountClient extends AbstractNetworkClient {
 
     public AccountClient(SDKClient sdkClient, RetryTemplate retryTemplate) {
         super(sdkClient, retryTemplate);
-        initialBalance = getBalance();
-        log.info("Operator account {} initial balance is {}", sdkClient.getExpandedOperatorAccountId(), initialBalance);
+        try {
+            initialBalance = getBalance();
+            log.info(
+                    "Operator account {} initial balance is {}",
+                    sdkClient.getExpandedOperatorAccountId(),
+                    initialBalance);
+        } catch (Throwable t) {
+            clean();
+            throw t;
+        }
     }
 
     @Override

--- a/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/SDKClient.java
+++ b/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/SDKClient.java
@@ -15,6 +15,7 @@ import com.hedera.hashgraph.sdk.TopicDeleteTransaction;
 import com.hedera.hashgraph.sdk.TopicId;
 import com.hedera.hashgraph.sdk.TopicMessageSubmitTransaction;
 import com.hedera.hashgraph.sdk.TransactionReceipt;
+import com.hedera.hashgraph.sdk.TransactionReceiptQuery;
 import com.hedera.hashgraph.sdk.proto.AccountID;
 import com.hedera.hashgraph.sdk.proto.NodeAddress;
 import com.hedera.hashgraph.sdk.proto.NodeAddressBook;
@@ -43,6 +44,7 @@ import org.hiero.mirror.test.e2e.acceptance.config.SdkProperties;
 import org.hiero.mirror.test.e2e.acceptance.props.ExpandedAccountId;
 import org.hiero.mirror.test.e2e.acceptance.props.NodeProperties;
 import org.hiero.mirror.test.e2e.acceptance.util.TestUtil;
+import org.springframework.retry.support.RetryTemplate;
 import org.springframework.util.CollectionUtils;
 
 @CustomLog
@@ -58,6 +60,7 @@ public class SDKClient implements Cleanable {
     private final AcceptanceTestProperties acceptanceTestProperties;
     private final SdkProperties sdkProperties;
     private final MirrorNodeClient mirrorNodeClient;
+    private final RetryTemplate retryTemplate;
     private final TopicId topicId;
 
     @Getter
@@ -69,21 +72,29 @@ public class SDKClient implements Cleanable {
     public SDKClient(
             AcceptanceTestProperties acceptanceTestProperties,
             MirrorNodeClient mirrorNodeClient,
+            RetryTemplate retryTemplate,
             SdkProperties sdkProperties,
             StartupProbe startupProbe)
             throws InterruptedException, TimeoutException {
-        defaultOperator = new ExpandedAccountId(
-                acceptanceTestProperties.getOperatorId(), acceptanceTestProperties.getOperatorKey());
-        this.mirrorNodeClient = mirrorNodeClient;
-        this.acceptanceTestProperties = acceptanceTestProperties;
-        this.sdkProperties = sdkProperties;
-        this.client = createClient();
-        var receipt = startupProbe.validateEnvironment(client);
-        this.topicId = receipt != null ? receipt.topicId : null;
-        validateClient();
-        expandedOperatorAccountId = getOperatorAccount(receipt);
-        this.client.setOperator(expandedOperatorAccountId.getAccountId(), expandedOperatorAccountId.getPrivateKey());
-        validateNetworkMap = this.client.getNetwork();
+        try {
+            defaultOperator = new ExpandedAccountId(
+                    acceptanceTestProperties.getOperatorId(), acceptanceTestProperties.getOperatorKey());
+            this.mirrorNodeClient = mirrorNodeClient;
+            this.acceptanceTestProperties = acceptanceTestProperties;
+            this.retryTemplate = retryTemplate;
+            this.sdkProperties = sdkProperties;
+            this.client = createClient();
+            var receipt = startupProbe.validateEnvironment(client);
+            this.topicId = receipt != null ? receipt.topicId : null;
+            validateClient();
+            expandedOperatorAccountId = getOperatorAccount(receipt);
+            this.client.setOperator(
+                    expandedOperatorAccountId.getAccountId(), expandedOperatorAccountId.getPrivateKey());
+            validateNetworkMap = this.client.getNetwork();
+        } catch (Throwable t) {
+            clean();
+            throw t;
+        }
     }
 
     public AccountId getRandomNodeAccountId() {
@@ -180,13 +191,22 @@ public class SDKClient implements Cleanable {
                         .getOperatorBalance()
                         .divide(exchangeRateUsd, 8, RoundingMode.HALF_EVEN));
 
-                var accountId = new AccountCreateTransaction()
+                var response = new AccountCreateTransaction()
                         .setAlias(alias)
                         .setInitialBalance(balance)
                         .setKeyWithoutAlias(publicKey)
-                        .execute(client)
-                        .getReceipt(client)
-                        .accountId;
+                        .execute(client);
+
+                // Verify all nodes have created the account since state is updated at different wall clocks
+                TransactionReceipt queryReceipt = null;
+                for (final var nodeAccountId : client.getNetwork().values()) {
+                    queryReceipt = retryTemplate.execute(x -> new TransactionReceiptQuery()
+                            .setNodeAccountIds(List.of(nodeAccountId))
+                            .setTransactionId(response.transactionId)
+                            .execute(client));
+                }
+
+                var accountId = queryReceipt.accountId;
                 log.info("Created operator account {} with public key {}", accountId, publicKey);
                 return new ExpandedAccountId(accountId, privateKey);
             }

--- a/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/TokenClient.java
+++ b/test/src/test/java/org/hiero/mirror/test/e2e/acceptance/client/TokenClient.java
@@ -547,7 +547,10 @@ public class TokenClient extends AbstractNetworkClient {
         KeyList keyList = KeyList.of(newTreasuryId.getPrivateKey());
         var response = executeTransactionAndRetrieveReceipt(tokenUpdateTransaction, keyList);
         log.info(
-                "Updated token {} treasury account {} via {}", tokenId, treasuryAccountId, response.getTransactionId());
+                "Updated token {} treasury account to {} via {}",
+                tokenId,
+                treasuryAccountId,
+                response.getTransactionId());
         return response;
     }
 

